bundles/org.eclipse.osgi/felix/src/org/apache/felix/resolver/util/ShadowList.java

```diff
@@ -48,8 +48,8 @@ private ShadowList(CandidateSelector shadow, List<Capability> original)
     }
 
     private ShadowList(List<Capability> unmodifiable, List<Capability> original, AtomicBoolean isUnmodifiable) {
-        super(unmodifiable, isUnmodifiable);
-        m_original = new ArrayList<Capability>(original);
+        super(new ArrayList<Capability>(unmodifiable), isUnmodifiable);
```

Comment:
I don't understand the switch here to copy for the super call and not for the `m_original`

Reply:
This code a a bit cumbersome because it switches "unmodifiable" what actually *is* modified later on but the original is later passed to the ProvisonContext.